sqldelight-idea-plugin/src/main/kotlin/app/cash/sqldelight/intellij/gradle/FileIndexMap.kt

```diff
@@ -26,11 +26,18 @@ import timber.log.Timber
 import java.io.File
 
 internal class FileIndexMap {
+  private var fetchThread: Thread? = null
   private val fileIndices = mutableMapOf<String, SqlDelightFileIndex>()
 
   private var initialized = false
   private var retries = 0
 
+  fun close() {
+    fetchThread?.interrupt()
```

Comment:
yes this is terrible - the gradle tooling `ProjectConnection.close` i'd like to call is hidden in the IntelliJ platform SDK so this is the only thing I could think of to stop the gradle invocation however if any gradle tooling gurus have better ideas I am all ears

Reply:
Ask in IntelliJ platform Slack?

Also what about attempting reflection on that API and falling back to thread interruption?